stacks-signer/src/signer.rs

```diff
@@ -1609,15 +1706,22 @@ fn load_encrypted_signer_state<S: SignerStateStorage>(
     storage: S,
     id: S::IdType,
     private_key: &Scalar,
-) -> Result<Option<v2::Signer>, PersistenceError> {
-    if let Some(encrypted_state) = storage.get_encrypted_signer_state(id)? {
-        let serialized_state = decrypt(private_key, &encrypted_state)?;
-        let state = serde_json::from_slice(&serialized_state)
-            .expect("Failed to deserialize decryoted state");
-        Ok(Some(v2::Signer::load(&state)))
-    } else {
-        Ok(None)
-    }
+) -> Result<StoredSignerStates, PersistenceError> {
+    let Some(encrypted_state) = storage.get_encrypted_signer_state(id)? else {
+        return Ok(VecDeque::new());
+    };
+    let serialized_state = decrypt(private_key, &encrypted_state)?;
+
+    Ok(serde_json::from_slice(&serialized_state)?)
+}
+
+fn get_signer_state(
+    loaded_signers: StoredSignerStates,
+    aggregate_key: Point,
+) -> Option<SignerState> {
+    loaded_signers
+        .into_iter()
+        .find(|state| state.group_key == aggregate_key)
```

Comment:
It seems like this method is only ever invoked once, and it's a pretty small function -- I think this function obscures more than it simplifies the code -- my suggestion is to just remove this function and move the code to `load_saved_state_for_aggregate_key`

Reply:
Done.